src/theory/quantifiers/oracle_engine.cpp

```diff
@@ -71,6 +77,30 @@ void OracleEngine::reset_round(Theory::Effort e) {}
 
 void OracleEngine::registerQuantifier(Node q) {}
 
+
+// need to know the oracle
+bool OracleEngine::checkConsistent(
+  const std::vector< std::pair<Node, Node> >& ioPairs, // input-output pairs from the oracle
+  std::vector<Node> &apps, std::vector<Node>& lemmas) // lemmas is to be filled, if the input-output pairs are inconsistent with model
+  {
+    bool consistent=true;
+    NodeManager* nm = NodeManager::currentNM();
+    eq::EqualityEngine* eq = getEqualityEngine();
+    for(unsigned int i=0; i<ioPairs.size(); i++)
+    {
+      // get predicted result
+      Node predictedResult = eq->getRepresentative(apps[i]);
```

Comment:
Actually, this isn't what I had wanted.

I wanted the input to contain the expected values; the method should call the oracles.

Reply:
So std::vector<Node, Node> ioPairs is the inputs (symbolic) and expected outputs? Or is it the inputs (values) and expected outputs?

Reply:
An example pair would be `f(5,6), 7`, return true iff the oracle thinks f(5,6)=7

Reply:
done